src/packages/StorageFileApi.ts

```diff
@@ -347,13 +355,61 @@ export default class StorageFileApi {
     }
   }
 
+  /**
+   * Download a file in a public bucket
+   *
+   * @param path The file path, including the current file name. For example `folder/image.png`.
+   * @param options.transform download a transformed variant of the image with the provided filters
+   */
+  publicDownload(
+    path: string,
+    options?: {
+      transform?: TransformOptions
+    }
+  ) {
+    const wantsTransformations = typeof options?.transform !== 'undefined'
+    const renderPath = wantsTransformations ? 'render/image' : 'object'
+    const queryString = this.transformOptsToQueryString(options?.transform || {})
+
+    return this.download(path, {
+      prefix: `${renderPath}/public`,
+      queryString: queryString,
+    })
```

Comment:
This is more of a question for the storage API server but is there an edge case here where if a user had a bucket named `public`, and they had a `cat.jpg` file in there, they were able to download the file with `${supabaseUrl}/storage/v1/object/public/cat.jpg` prior to this update, but now they cannot?